### DISCUSSION TITLE
Equation numbering w/o labels

### DISCUSSION THREAD

**Discussion by agerlach** [quarto-dev/quarto-cli#4132] [Q&A]
Is is possible to add equation numbers for all `$$...$$` equations w/o adding a label?

For example, I would like the following to still number the equations 1, 2, 3 respectively. Currently a is number 1, b is not numbered and c is numbered 3.

````markdown
$$a$$ {#eq-a}

$$b$$

$$c$$ {#eq-c}
````

**Comment by eeholmes** [comment]
If you use `$$ $$` format (display math), then to have equation numbers, you need to add `{#eq-xyp}` on every equation. It is just the nature of `$$ $$`.

**Comment by agerlach** [comment]
What other options is there outside of $$ $$ ?

**Comment by eeholmes** [comment]
You can use what is called LaTeX equation environments. However, learning LaTeX is probably overkill. But here is how you do it:

```
\begin{equation}\tag{1}\label{eq-a}
a
\end{equation}

\begin{equation*}
b
\end{equation*}

\begin{equation}\tag{3}\label{eq-b}
c
\end{equation}
```
You have to set fixed tags, not autonumbered, because you are telling it to skip ahead to eq 3 essentially.

Note for html you have to turn on equation numbering if you use the math environment. For PDF output, you don't have to do this.

To do this, create a file called mathjax.html (say) with this:
```
<script>
  MathJax = {
    tex: {
      tags: 'ams'  // should be 'ams', 'none', or 'all'
    }
  };
</script>
```
And add this to your html header with
```
---
title: "test"
format:
  html:
    include-in-header: mathjax.html
---
```

**Comment by baptiste** [comment]
One could probably write a lua filter that would detect if an equation is unlabelled, and add an automatic label. [This filter](https://github.com/quarto-dev/quarto-cli/blob/56da834f07f5fdfab1e432f11aa3be6b26f4fd2a/src/resources/filters/crossref/equations.lua) might be a good starting point (I wasn't able to adapt it in my brief attempt).

**Comment by eeholmes** [comment]
@baptiste Do you know how to tell Quarto to use a modified equations.lua instead of this one? I know how to replace templates and template-partials but not lua filters.

**Reply by baptiste** [reply]
Not sure, sorry; usually filters just act independently one after the other, but here there's probably a risk of making incompatible/duplicated changes to the AST, I guess.

**Comment by jtlandis** [comment]
I wanted to do the same thing as @agerlach. Check out the [quarto filter](https://github.com/jtlandis/auto-eq) I made which is adapted from the resource @baptiste found.

It loosely works, but please let me know if you find any bugs
